lhotse/workflows/activity_detection/base.py

```diff
@@ -0,0 +1,126 @@
+import abc
+import multiprocessing
+from concurrent.futures import ProcessPoolExecutor
+from dataclasses import dataclass
+from functools import partial
+from itertools import chain
+from typing import Dict, List, Optional, Type
+
+import numpy as np
+
+from lhotse.audio.recording import Recording
+from lhotse.audio.recording_set import RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+
+
+@dataclass
+class Activity:
+    start: float
+    duration: float
+
+
+class ActivityDetector(abc.ABC):
+    def __init__(self, detector_name: str, sampling_rate: int, device: str = "cpu"):
+        self._detector_name = detector_name
+        self._sampling_rate = sampling_rate
+        self._device = device
+
+    @property
+    def device(self) -> str:
+        return self._device
+
+    def __call__(self, recording: Recording) -> List[SupervisionSegment]:
+        resampled = recording.resample(self._sampling_rate)
+        audio = resampled.load_audio()  # type: ignore
+
+        uid_template = "{recording_id}-{detector_name}-{channel}-{number:05}"
+
+        result: List[SupervisionSegment] = []
+        for channel, track in enumerate(audio):
+            track = np.squeeze(track)
+            activities = self.forward(track)
+
+            for i, activity in enumerate(activities):
+                uid = uid_template.format(
+                    recording_id=recording.id,
+                    detector_name=self._detector_name,
+                    channel=channel,
+                    number=i,
+                )
+                segment = SupervisionSegment(
+                    id=uid,
+                    recording_id=recording.id,
+                    start=activity.start,
+                    duration=activity.duration,
+                    channel=channel,
```

Comment:
FYI you could leverage a custom field here to tag which vad created the segment, i.e.:

```python
SupervisionSegment(
    ...,
    custom={"vad_tag": detector_name},
)
```

but the current solution is also OK with me